src/data/detail-page/charts/chartInterfaces.ts

```diff
@@ -1,4 +1,17 @@
 export interface Chart {
-    data?: string[];
-    type?: string;
+  data?: ChartDataItem[];
```

Comment:
We have multiple optional properties on this interface and this is not a good sign - if we make everything optional we're giving up much of the protection we get from TypeScript. 
Let's reconsider this interface: it's unlikely that we will ever have a chart without `data` or `type` so we can probably make them both non-optional. For the others, we can make them `string | null` or `Source | null` and remove the optional `?`. That way we will have to explicitly set them to null if we want to skip them (if ever).

Reply:
Chart interface is now slightly refactored.